### PR TITLE
Miner Enchant Fix

### LINKED ISSUE #2568: miner doesnt use specfic enchants
### Minecolonies version
1.12.2-0.8.9630

### Expected behavior
- to use the enchants as normal

### Actual behaviour
- completely ignores the silk touch and fortune enchants from both tinkers construct and vanilla

### Steps to reproduce the problem
craft any pick enchant them with either silk touch or fortune 1-3 give to miner and see him ignore the enchants

### PR DESCRIPTION
**Pull request by ghost**
Closes #2568

# Changes proposed in this pull request:
-Tested Fortune Enchant.(Both TiCon Tools and Vanilla) Fortune was implemented successfully during miner re-writes, but was never made note of.
- Added hasSilkTouch Boolean to ItemStackUtils that returns true if Silk Touch is on item.
-Implemented SilkTouch test and item grab in mineBlock function in AbstractEntityAiInteract.
-Tested Silk Touch Enchants(Both TiCon and Vanilla) successfully

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIInteract.java [hunk 1]
```diff
@@ -157,8 +158,24 @@ protected final boolean mineBlock(@NotNull final BlockPos blockToMine, @NotNull
         //calculate fortune enchantment
         final int fortune = ItemStackUtils.getFortuneOf(tool);
 
-        //get all item drops
-        final List<ItemStack> localItems = BlockPosUtil.getBlockDrops(world, blockToMine, fortune);
+        //check if tool has Silk Touch
+        final boolean silkTouch = ItemStackUtils.hasSilkTouch(tool);
+
+        //create list for all item drops to be stored in
+        List<ItemStack> localItems = new ArrayList<ItemStack>();
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIInteract.java]: I think we can leave this list final and only do addAll in the logic down there.

File: src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIInteract.java [hunk 2]
```diff
@@ -263,8 +280,9 @@ public int getBlockMiningDelay(@NotNull final Block block, @NotNull final BlockP
             return (int) world.getBlockState(pos).getBlockHardness(world, pos);
         }
 
-        return (int) ((Configurations.gameplay.pvp_mode ? Configurations.gameplay.blockMiningDelayModifier/2 : Configurations.gameplay.blockMiningDelayModifier
-                         * Math.pow(LEVEL_MODIFIER, worker.getCitizenExperienceHandler().getLevel()))
+        return (int) ((Configurations.gameplay.pvp_mode ? Configurations.gameplay.blockMiningDelayModifier / 2 : Configurations.gameplay.blockMiningDelayModifier
+                                                                                                                   * Math.pow(LEVEL_MODIFIER,
+          worker.getCitizenExperienceHandler().getLevel()))
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIInteract.java]: Seems like the formatter derped here

**Comment by Raycoms** [comment]
Looks fine besides the two small things I pointed out.

**Comment by ghost** [comment]
Fixed

**Pull request merged**